DuckDuckGo/TabsModel.swift

```diff
@@ -152,4 +152,14 @@ public class TabsModel: NSObject, NSCoding {
         tabs.append(Tab())
         currentIndex = 0
     }
+    
+    func tabExists(withHost host: String) -> Bool {
+        return !tabs.filter {
+            if let tabHost = $0.link?.url.host {
+                return tabHost == host
+            }
+
+            return false
+        }.isEmpty
```

Comment:
We could speed this up using `contains`, since the current implementation will filter the entire tab list every time, even when the tab it's looking for it one of the first in the list. `contains` on the other hand should just stop checking once it finds a match:

```suggestion
        return tabs.contains { $0.link?.url.host == host }
```

Reply:
Thanks for the suggestion! I've made that change + tested